src/ControlSystem/ControlErrors/Rotation.hpp

```diff
@@ -81,21 +85,23 @@ struct Rotation : tt::ConformsTo<protocols::ControlError> {
            "Excision sphere for ObjectB not in the domain but is needed to "
            "compute Rotation control error.");
 
-    const DataVector grid_position_of_A = array_to_datavector(
-        domain.excision_spheres().at("ObjectAExcisionSphere").center());
-    const DataVector grid_position_of_B = array_to_datavector(
-        domain.excision_spheres().at("ObjectBExcisionSphere").center());
-    const DataVector& current_position_of_A = get<center_A>(measurements);
-    const DataVector& current_position_of_B = get<center_B>(measurements);
+    const auto& grid_position_of_A =
+        domain.excision_spheres().at("ObjectAExcisionSphere").center();
+    const auto& grid_position_of_B =
+        domain.excision_spheres().at("ObjectBExcisionSphere").center();
+    const auto& current_position_of_A = get<center_A>(measurements);
+    const auto& current_position_of_B = get<center_B>(measurements);
 
     // A is to the right of B in grid frame. To get positive differences,
     // take A - B
-    const DataVector grid_diff = grid_position_of_A - grid_position_of_B;
-    const DataVector current_diff =
-        current_position_of_A - current_position_of_B;
+    const auto grid_diff_tnsr = tenex::evaluate<ti::I>(
+        grid_position_of_A(ti::I) - grid_position_of_B(ti::I));
+    const DataVector grid_diff{
+        {grid_diff_tnsr[0], grid_diff_tnsr[1], grid_diff_tnsr[2]}};
+    const auto current_diff = current_position_of_A - current_position_of_B;
 
-    const double grid_dot_current = dot(grid_diff, current_diff);
-    const DataVector grid_cross_current = cross(grid_diff, current_diff);
+    const auto grid_dot_current = dot(grid_diff, current_diff);
+    const auto grid_cross_current = cross(grid_diff, current_diff);
```

Comment:
Personally I think the explicit types that you removed here were helpful to understand this code, in particular because you're dealing with DataVector, Tensor, double, etc and it's not obvious from reading the code what the type is. Your IDE knows it, but GitHub doesn't. I'd say keep the explicit types. Same in the other files.

Reply:
okay, it's one of Scott Meyer's rules to pretty much always prefer auto, also because it makes these refactors a lot easier. But I don't care that much.

Reply:
We've followed that rule for a long time and it's still the default, but we have started to prefer explicit return types where it makes the code easier to understand.